Doc improvements: typos, grammar, formatting (#1351)
accuknox/KubeArmor: getting-started/FAQ.md

</details>

<details><summary><h4>How is KubeArmor different from PodSecurityPolicy/PodSecurityContext?</h4></summary>

Native k8s support specifying a security context for the pod or container. It requires one to specify native AppArmor, SELinux, and seccomp policies. But there are a few problems with this approach:  

* All the OS distributions do not support the LSMs consistently. For e.g, [GKE COS](https://cloud.google.com/container-optimized-os/) supports AppArmor while [Bottlerocket](https://aws.amazon.com/bottlerocket/) supports SELinux and BPF-LSM.  
* The Pod Security Context expects the security profile to be specified in its native language, for instance, the AppArmor profile for AppArmor. SELinux profile if SELinux is to be used. The profile language is extremely complex and this complexity could backfire i.e., it could lead to security holes.  
* Security Profile updates are manual and difficult: When an app is updated, the security posture might change and it becomes difficult to manually update the native rules.  

> * No alerting of LSM violation on managed cloud platforms: By default, LSMs send logs to kernel audit, which is not available on most managed cloud platforms.  

Reviewer: It should be auditd (i.e. audit daemon)

KubeArmor solves all the above-mentioned problems.

* It maps YAML rules to LSMs (apparmor, bpf-lsm) rules so prior knowledge of different security contexts (native AppArmor, SELinux) is not required.  
* It's easy to deploy: KubeArmor is deployed as a daemon set. Even when the application is updated, the enforcement rules are automatically applied.  
* Consistent Alerting: KubeArmor handles kernel events and maps k8s metadata using ebpf.  
* KubeArmor also runs in systemd mode so can directly run and protect Virtual Machines or Bare-metal machines too.  
* Pod Security Context cannot leverage BPF-LSM at all today. BPF-LSM provides more programmatic control over the policy rules.  

> * Pod Security Context does not manage abstractions. As an example, you might have two nodes with Ubuntu, and two nodes with Bottlerocket. Ubuntu, by default, has AppArmor and Bottlerocket has BPF-LSM and SELinux. KubeArmor internally picks the right primitives to use for enforcement and the user does not have to bother explicitly stating what to use.

Reviewer: ```suggestion
* The Pod Security Context does not handle abstractions. For instance, consider a scenario where you have a node running Ubuntu and another node running Bottlerocket. Ubuntu inherently utilizes AppArmor, while Bottlerocket employs BPF-LSM and SELinux as its default LSMs. KubeArmor internally selects the appropriate primitives for enforcement, eliminating the need for users to explicitly specify the enforcement method.
```

  "Result": "Permission denied"
}
```

Here the log implies that the process /usr/bin/sleep execution by 'zsh' was denied on the Host using a block-based host policy.

The logs are also exportable in OpenTelemetry format using [kubearmor/OTel-receiver](https://github.com/kubearmor/OTel-receiver).

</details>

<details><summary><h4>How to visualize KubeArmor visibility logs</h4></summary>

There are a couple of community-maintained dashboards available at [kubearmor/kubearmor-dashboards](https://github.com/kubearmor/kubearmor-dashboards).

If you don't find an existing dashboard particular to your needs, feel free to create an issue. It would be great if you could also contribute one!
</details>

<details><summary><h4>How to get process events in the context of specific pods?</h4></summary>  

The following command can be used to get pod specific events:  

> `karmor log --pod <pod_name>`  

Reviewer: ```suggestion
`karmor logs --pod <pod_name>`  
```

> `karmor log` has the following filter to provide more granularity:

Reviewer: ```suggestion
`karmor logs` has the following filter to provide more granularity:
```

Please update `karmor log` to `karmor logs` at other places too

  UEK R7 can be installed on OL 8.6 by following the easy-to-follow instructions provided here in this [Oracle Blog Post](https://blogs.oracle.com/post/uek-7-oracle-linux-8).

> Note: After upgrading to the UEK R7 you may required to enable BPF-LSM if it's not enabled by default.

### Checking if BPF-LSM is enabled

- check if bpf is enabled by verifying if it is in the active lsms.
* check if bpf is enabled by verifying if it is in the active lsms.

  ```
  $ cat /sys/kernel/security/lsm
  capability,yama,selinux,bpf
  ```

  as we can see here `bpf` is in active lsms

### Enabling BPF-LSM manually using boot configs

- Open the `/etc/default/grub` file in privileged mode.
* Open the `/etc/default/grub` file in privileged mode.

  ```

>   $ sudo vi /etc/default/grub

Reviewer: Good find!
